### DISCUSSION TITLE
how to fix this please Cannot resolve method 'terminateApp' in 'AppiumDriver'

### DISCUSSION THREAD

**Discussion by HamzaMiz** [appium/appium#18014] [Q&A] [answered]
hello ,
my problem is not urgent but dont know why the terminateApp and activateApp are not called when i do driver.terminateApp 
<img width="829" alt="Screenshot 2023-01-05 at 15 36 34" src="https://user-images.githubusercontent.com/94066399/210804950-93756d13-a4fb-4789-ab94-e748be3de5f0.png">

right now i m use this method to those action

 static void activateApp(){
        Map<String, Object> args = new HashMap<>();
        args.put("bundleId", "com.niji.totalwaykonect.myvehiculee.dev");
        args.put("appId", " com.totalenergies.myvehicle.dev");
        ((JavascriptExecutor) driver).executeScript("mobile:activateApp", args);
    }

my driver is declared like this     public static AppiumDriver driver = null;

my pom
<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>cucumber-testng</artifactId>
			<version>7.3.3</version>
			<scope>compile</scope>
		</dependency>
		<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>cucumber-core</artifactId>
			<version>${cucumber.version}</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>cucumber-java</artifactId>
			<version>${cucumber.version}</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>com.github.appium</groupId>
			<artifactId>java-client</artifactId>
			<version>8.3.0</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>io.cucumber</groupId>
			<artifactId>cucumber-picocontainer</artifactId>
			<version>${cucumber.version}</version>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.seleniumhq.selenium</groupId>
			<artifactId>selenium-firefox-driver</artifactId>
			<version>${selenium.version}</version>
		</dependency>
		<dependency>
			<groupId>org.seleniumhq.selenium</groupId>
			<artifactId>selenium-chrome-driver</artifactId>
			<version>${selenium.version}</version>
		</dependency>
		<dependency>
			<groupId>org.seleniumhq.selenium</groupId>
			<artifactId>selenium-support</artifactId>
			<version>${selenium.version}</version>
			<scope>test</scope>

		</dependency>
		<dependency>
			<groupId>org.seleniumhq.selenium</groupId>
			<artifactId>selenium-api</artifactId>
			<version>${selenium.version}</version>
			<scope>test</scope>

		</dependency>
		<dependency>
			<groupId>org.seleniumhq.selenium</groupId>
			<artifactId>selenium-java</artifactId>
			<version>${selenium.version}</version>
			<scope>test</scope>

		</dependency>

i feel something is missing , any help will be appreciate it

**Comment by KazuCocoa** [comment, marked as answer]
Maybe the driver should be AndroidDriver or iOSDriver?

**Reply by HamzaMiz** [reply]
Hello , thank u the cast worked , but does MobileDriver works for both ?

**Reply by KazuCocoa** [reply]
The availability is implements as https://github.com/appium/java-client/search?q=InteractsWithApps
